src/lib/server/embeddingEndpoints/embeddingEndpoints.ts

```diff
@@ -0,0 +1,38 @@
+import {
+	embeddingEndpointTei,
+	embeddingEndpointTeiParametersSchema,
+} from "./tei/embeddingEndpoints";
+import { z } from "zod";
+import embeddingEndpointXenova, {
+	embeddingEndpointXenovaParametersSchema,
+} from "./xenova/embeddingEndpoints";
+
+// parameters passed when generating text
+interface EmbeddingEndpointParameters {
+	inputs: string[];
+}
+
+interface CommonEmbeddingEndpoint {
+	weight: number;
+}
+
+// type signature for the endpoint
+export type EmbeddingEndpoint = (params: EmbeddingEndpointParameters) => Promise<number[][]>;
+
+// generator function that takes in parameters for defining the endpoint and return the endpoint
+export type EmbeddingEndpointGenerator<T extends CommonEmbeddingEndpoint> = (
+	parameters: T
+) => EmbeddingEndpoint;
+
+// list of all endpoint generators
+export const embeddingEndpoints = {
+	tei: embeddingEndpointTei,
+	xenova: embeddingEndpointXenova,
```

Comment:
I think you can type it even stronger here. It can look something like:
```ts
const EMBEDDING_ENDPOINT_TYPES = ["tei", "xenova"];
type EmbeddingEndpointType = typeof EMBEDDING_ENDPOINTS[number];
export const embeddingEndpoints: Record<EmbeddingEndpointType, ...> = {
	tei: embeddingEndpointTei,
	xenova: embeddingEndpointXenova,
}

```